### PR TITLE
Rename ignore label and return error on controlplane failure

### PR DESCRIPTION
**Pull request by mrajashree**
https://github.com/rancher/rancher/issues/25606
https://github.com/rancher/rancher/issues/25599

### CODE REVIEW

File: services/node_util.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func resetMaxUnavailable(maxUnavailable, lenInactiveHosts int, component string)
 	}
 
 	if lenInactiveHosts > 0 {
-		if maxUnavailable == lenInactiveHosts {
+		if lenInactiveHosts >= maxUnavailable {
 			return 0, fmt.Errorf("cannot proceed with upgrade of %s since %v host(s) are found to be inactive prior to upgrade", component, lenInactiveHosts)
```
Comment (deniseschannon) [inline services/node_util.go]: Nit: Is inactive the way we want to describe it? There are so many reasons why a host would not be available to upgrade. Or what if the upgrade failed?

File: services/node_util.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int) (int, error
 		logrus.Errorf("Unable to parse max_unavailable, should be a number or percentage of nodes, error: %v", err)
```
Comment (deniseschannon) [inline services/node_util.go]: Do we check if it's negative and error out?

File: services/node_util.go [hunk 3]
```diff
@@ -79,25 +80,38 @@ func getNodeListForUpgrade(kubeClient *kubernetes.Clientset, hostsFailed *sync.M
...
 }
 
-func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int) (int, error) {
+func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int, role string) (int, error) {
 	// if maxUnavailable is given in percent, round down
 	maxUnavailableParsed := k8sutil.Parse(maxUnavailableVal)
 	logrus.Debugf("Provided value for maxUnavailable: %v", maxUnavailableParsed)
+	if maxUnavailableParsed.Type == k8sutil.Int {
+		if maxUnavailableParsed.IntVal <= 0 {
+			return 0, fmt.Errorf("invalid input for max_unavailable_%s: %v, value must be >= 1", role, maxUnavailableParsed.IntVal)
```
Comment (deniseschannon) [inline services/node_util.go]: Should it just be > 0 since it could be an int or percentage?

File: services/node_util.go [hunk 4]
```diff
@@ -79,25 +80,38 @@ func getNodeListForUpgrade(kubeClient *kubernetes.Clientset, hostsFailed *sync.M
...
+		}
+	} else {
+		if strings.HasPrefix(maxUnavailableParsed.StrVal, "-") {
```
Comment (deniseschannon) [inline services/node_util.go]: Can this just be consolidated to check if it's negative or 0 and just ask that it's >0 as the error message?

File: services/node_util.go (hunk 1 above)
Comment (deniseschannon) [reply services/node_util.go]: Could we update to indicate what "inactive" means in this case?

File: services/node_util.go (hunk 3 above)
Comment (mrajashree) [reply services/node_util.go]: I'm checking separately for percentage type, `maxUnavailableParsed.IntVal` won't hold any percentage values

File: services/node_util.go [hunk 5]
```diff
@@ -79,25 +80,35 @@ func getNodeListForUpgrade(kubeClient *kubernetes.Clientset, hostsFailed *sync.M
...
+func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int, role string) (int, error) {
 	// if maxUnavailable is given in percent, round down
 	maxUnavailableParsed := k8sutil.Parse(maxUnavailableVal)
 	logrus.Debugf("Provided value for maxUnavailable: %v", maxUnavailableParsed)
+	if maxUnavailableParsed.Type == k8sutil.Int {
+		if maxUnavailableParsed.IntVal <= 0 {
+			return 0, fmt.Errorf("invalid input for max_unavailable_%s: %v, value must be > 0", role, maxUnavailableParsed.IntVal)
+		}
+	} else {
+		if strings.HasPrefix(maxUnavailableParsed.StrVal, "-") && maxUnavailableParsed.StrVal == "0%" {
```
Comment (deniseschannon) [inline services/node_util.go]: shouldn't this be an OR?

File: services/node_util.go [hunk 6]
```diff
@@ -79,25 +80,35 @@ func getNodeListForUpgrade(kubeClient *kubernetes.Clientset, hostsFailed *sync.M
...
+			return 0, fmt.Errorf("invalid input for max_unavailable_%s: %v, percentage input should be > 0", role, maxUnavailableParsed.StrVal)
```
Comment (deniseschannon) [inline services/node_util.go]: Small Nit: But in theory, it should have the same message as above to say `value must be > 0`

**Comment by deniseschannon** [comment]
https://github.com/rancher/rke/issues/1906

**Pull request merged**